pkg/controller/seed-controller-manager/cluster-template-controller/controller.go

```diff
@@ -242,11 +242,19 @@ func (r *reconciler) assignSSHKeyToCluster(ctx context.Context, clusterID string
 func genNewCluster(template *kubermaticv1.ClusterTemplate, instance *kubermaticv1.ClusterTemplateInstance, workerName string) *kubermaticv1.Cluster {
 	name := utilcluster.MakeClusterName()
 
+	annotations := make(map[string]string)
+	for key, val := range template.ClusterAnnotations {
+		annotations[key] = val
+	}
+	for key, val := range template.Annotations {
+		annotations[key] = val
+	}
+
 	newCluster := &kubermaticv1.Cluster{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        name,
 			Labels:      template.ClusterLabels,
-			Annotations: template.Annotations,
+			Annotations: annotations,
```

Comment:
Given this, why can't the desired annotations be kept in template.Annotations? Why do we need a new field? At first glance without context it seems to do the same as the existing field in the template's metadata?

Reply:
@yaa110 any insights?

Reply:
Did you mean to mutate `template.Annotations`? Here, `template.Annotations` and `ClusterAnnotations` are merged.

Reply:
I'm unsure why this PR was merged; I have the same question as Christoph. Why not simply use ClusterTemplate.Annotations? Since you are already changing how annotations are propagated in your PR. 

`ClusterLabels` is a dedicated field for labels since it isolates which labels will be propagated to the Clusters created against a ClusterTemplate. If you are introducing the field`ClusterAnnotations`, I'd expect the same behavior there as well. 

The behavior after your PR is that all the values from `ClusterTemplate. ClusterAnnotations` and `ClusterTemplate.Annotations` will be propagated to the Clusters with somehow `ClusterTemplate.Annotations` having a higher precedence than the dedicated field that you introduced, `ClusterTemplate. ClusterAnnotations`.

Reply:
I approved it because I felt it was hanging and wanted to allow @yaa110 to progress and iterate on it. But you're right, I probably shouldn't have.